app/src/main/java/org/fossasia/phimpme/gallery/activities/LFMainActivity.java

```diff
@@ -2042,13 +2044,23 @@ public void onClick(DialogInterface dialog, int which) {
 
           private boolean succ = false;
           private int imagesUnfav = 0;
+          private Dialog dialog;
 
           @Override
           protected void onPreExecute() {
+            dialog = getLoadingDialog(context, "Deleting", false);
+            dialog.show();
             swipeRefreshLayout.setRefreshing(true);
             super.onPreExecute();
           }
 
+          private Dialog getLoadingDialog(Context context, String title, boolean canCancel) {
+            ProgressDialog dialog = new ProgressDialog(context);
```

Comment:
ProgressDialog was depreciated in api 26 as i remember. Use progress bar instead

Reply:
Is it necessary? ProgressDialog is still working.

Reply:
Deprecated doesn't mean disabled

Reply:
@iamareebjamal I know. I think that we shouldn't shift to Progress bar because everywhere we are using ProgressDialog. If we shift this to progress bar, we should shift others too to progress bars.

Reply:
> A program element annotated @Deprecated is one that programmers are discouraged from using, typically because it is dangerous, or because a better alternative exists.


It's upto you, but it is suggested as it is more robust to api changes

Reply:
Yeah, not everytime you can have a UI component which lets user dismiss it. See the stackoverflow thread that not everyone is happy with this change,